### PR TITLE
Fix edge cases

### PR DESCRIPTION
**Pull request by yanovs**
- Fix `__contains__`
- Fix more complex perturb case
- Add `publish-test` to CI
- Simplify CI by switching from [`tox-gh-actions`](https://github.com/ymyzk/tox-gh-actions) to `tox` labels ([available in `tox >= 4.0.0`](https://tox.wiki/en/latest/faq.html#new-features-in-tox-4))
- Add `py.typed`

### CODE REVIEW

File: dilib/utils.py [hunk 1]
```diff
@@ -56,6 +56,40 @@ def nested_getattr(obj: Any, address: str) -> Any:
     return obj
 
 
+def nested_contains(obj: Any, address: str) -> Any:
```
Comment (adshieh) [inline dilib/utils.py:59]: This returns a `bool`.

File: dilib/container.py [hunk 2]
```diff
@@ -51,23 +54,31 @@ def config(self) -> TC:
         # Cast because ConfigProxy[TC] will act like TC
         return cast(TC, ConfigProxy(self, self._config))
 
+    # noinspection PyProtectedMember
```
Comment (adshieh) [inline dilib/container.py:57]: What do you think about making `config._keys` part of the public interface, so we don't need to suppress these PyCharm warnings?

File: dilib/utils.py (hunk 1 above)
Comment (yanovs) [reply dilib/utils.py]: Fixed.

File: dilib/container.py (hunk 2 above)
Comment (yanovs) [reply dilib/container.py:57]: I'm not sure--on the one hand, I've thought about adding "meta" helpers that expose the delayed spec structure, but on the other hand, I'm not sure what the use case is for that, and I wanted to keep the public interface as small as possible. I thought of these internal `Config` indices as specifically designed for the "friend" `Container` class to use.
